clang-tools-extra/clang-tidy/bugprone/PublicEnableSharedFromThisCheck.cpp

```diff
@@ -0,0 +1,45 @@
+//===--- PublicEnableSharedFromThisCheck.cpp - clang-tidy -------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PublicEnableSharedFromThisCheck.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::bugprone {
+
+  void PublicEnableSharedFromThisCheck::registerMatchers(MatchFinder *match_finder) {
+      match_finder->addMatcher(
+              cxxRecordDecl(
+                  hasAnyBase(
```

Comment:
`hasAnyBase(cxxBaseSpecifier(...).bind("base"))`, check if this works, maybe with that you could avoid having loop in a check method

Reply:
![image](https://github.com/user-attachments/assets/e5c1716a-ad63-441c-959a-f9a5d6717285)
I've checked, the binding comes up empty handed (see that there's no hello2 output in the terminal)

Unless you have a better idea to make it not come up empty?